### ISSUE TITLE
.NET 8 SDK "Object reference not set..." when trying to use template

### ISSUE THREAD

**Issue by akarnokd** [BepInEx/BepInEx.Templates#8]
With .NET 8 out, the usual way of using the template results in a not too detailed error message.

Can you look into the compatibility issue?

# Issue

```
C:\temp\1>dotnet new bepinex5plugin -n MyModNameHere -T netstandard2.1 -U 2023.1.9
Template "BepInEx 5 Plugin Template" could not be created.
Failed to create template.
Details: Object reference not set to an instance of an object.


For details on the exit code, refer to https://aka.ms/templating-exit-codes#100
```

Yes, the template (1.4) is installed:

```
C:\temp\2>dotnet new -i BepInEx.Templates --nuget-source https://nuget.bepinex.dev/v3/index.json
Warning: use of 'dotnet new --install' is deprecated. Use 'dotnet new install' instead.
For more information, run:
   dotnet new install -h

The following template packages will be installed:
   BepInEx.Templates

BepInEx.Templates (version 1.4.0) is already installed, it will be replaced with latest version.
BepInEx.Templates::1.4.0 was successfully uninstalled.
Success: BepInEx.Templates::1.4.0 installed the following templates:
Template Name                            Short Name            Language  Tags
---------------------------------------  --------------------  --------  --------------------------------------
BepInEx 5 Plugin Template                bepinex5plugin        [C#]      BepInEx/BepInEx 5/Plugin
BepInEx 6 .NET Launcher Plugin Template  bep6plugin_netfx      [C#]      BepInEx/BepInEx 6/Plugin/.NET Launcher
BepInEx 6 Il2Cpp Plugin Template         bep6plugin_il2cpp     [C#]      BepInEx/BepInEx 6/Plugin/Il2Cpp
BepInEx 6 Unity Mono Plugin Template     bep6plugin_unitymono  [C#]      BepInEx/BepInEx 6/Plugin/Unity Mono
```

# Workaround

- Install .NET 7 (7.0.404)
- Force use 7 in the target directory via `dotnet new globaljson --sdk-version 7.0.404`
- Issue the create code again

```
C:\temp\1>dotnet new bepinex5plugin -n MyModNameHere -T netstandard2.1 -U 2023.1.9

Welcome to .NET 7.0!
---------------------
SDK Version: 7.0.404

Telemetry
---------
The .NET tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
The template "BepInEx 5 Plugin Template" was created successfully.
```

**Comment by ManlyMarco** [comment]
Shouldn't this be reported to .NET as a regression?

**Comment by akarnokd** [comment]
Alright, let's the :point_right: :point_left: begin.

https://github.com/dotnet/sdk/issues/36991

**Comment by patrickdundas** [comment]
@akarnokd I ran into this issue earlier today and resolved it by installing .NET 6 and uninstalling .NET 8 via Windows "Apps & Features" Visual Studio Modify Installation button, which allows you to uncheck .NET 8 to uninstall it via the Visual Studio Installer. Once this is uninstalled, you will see .NET 6 when running `dotnet --version`. Once that is done, use the `dotnet new` CLI to create from the template like usual. You may need to restart your computer / Visual Studio.

**Comment by akarnokd** [comment]
@patrickdundas I already had the workaround and wasn't looking for one. The issue at hand is why it broke with .NET 8 and if it's the template's fault or .NET 8's. Apparently, it's a bug with .NET 8.

**Comment by Rubyboat1207** [comment]
For anyone who does not want to uninstall .NET 8 to make the project. Here's the steps I used:

1. run `dotnet --list-sdks`
Here's what mine looked like:
```
6.0.417 [C:\Program Files\dotnet\sdk]
7.0.304 [C:\Program Files\dotnet\sdk]
7.0.400 [C:\Program Files\dotnet\sdk]
8.0.100-preview.7.23376.3 [C:\Program Files\dotnet\sdk]
```

2. make this file in the root of your current directory:
```json
{
    "sdk": {
        "version": "<dotnet 6 or 7 version>"
    }
}
```
mine looked like this:
```
{
    "sdk": {
        "version": "7.0.400"
    }
}
```

3. then run the `dotnet new` as you would otherwise.

Hope this helps anyone

**Comment by fhgaha** [comment]
Instructions unclear.
> 2. make this file and call it global.json in your current directory:
what is current directory? directory where i try to create my mod? shouldnt global.json be somewhere in dotnet directory?
> 3. then run the dotnet new as you would otherwise.
You mean these commands
`dotnet new bep6plugin_unity_mono -n MyFirstPlugin -T <TFM> -U <Unity>
dotnet restore MyFirstPlugin`
from [bepinex guide]( https://docs.bepinex.dev/master/articles/dev_guide/plugin_tutorial/2_plugin_start.html)?

**Comment by LittleSaya** [comment]
> Instructions unclear.
> 
> > 2. make this file and call it global.json in your current directory:
> 
> what is current directory? directory where i try to create my mod? shouldnt global.json be somewhere in dotnet directory?
> 
> > 3. then run the dotnet new as you would otherwise.
> 
> You mean these commands from [bepinex guide](https://docs.bepinex.dev/master/articles/dev_guide/plugin_tutorial/2_plugin_start.html)? `dotnet new bep6plugin_unity_mono -n MyFirstPlugin -T <TFM> -U <Unity>` `dotnet restore MyFirstPlugin`

I found a document about `global.json`

**Comment by Rubyboat1207** [comment]
> Instructions unclear.
> 
> > 2. make this file and call it global.json in your current directory:
> 
> what is current directory? directory where i try to create my mod? shouldnt global.json be somewhere in dotnet directory?
> 
> > 3. then run the dotnet new as you would otherwise.
> 
> You mean these commands from [bepinex guide](https://docs.bepinex.dev/master/articles/dev_guide/plugin_tutorial/2_plugin_start.html)? `dotnet new bep6plugin_unity_mono -n MyFirstPlugin -T <TFM> -U <Unity>` `dotnet restore MyFirstPlugin`


"current directory" meant wherever you planned on running dotnet new. so you could call that the root of your project folder

and yes i do mean that command.

**Comment by jfmherokiller** [comment]
for anyone coming upon this issue the simplest method is using the global.json file. 

and unless there is a way to set it via some kind of environment variable or the like you must create it in the same directory you wish to store the project.

What that means is you want the file in the directory you are going to run a command like `dotnet new bepinex5plugin -n OuterWildsTest` 

if you put the file in the correct location the command above will succeed and the resulting directory will look like 

```
 OuterWildsTest
 global.json
 ```

**Comment by ManlyMarco** [comment]
Was this resolved by merging #12?

**Comment by ManlyMarco** [comment]
That sounds like a different issue, closing this as resolved then.
